src/react/hooks/auth.ts

```diff
@@ -1,6 +1,35 @@
 import { USER } from "../../apollo/queries";
+import {
+  Maybe,
+  UserFragment,
+  UserDetailsQuery,
+  UserDetailsQueryVariables,
+} from "../../apollo/types";
 import { hookFactory } from "../helpers/hookFactory";
 import { hookStateFactory } from "../helpers/hookStateFactory";
 
+type AuthState = {
```

Comment:
I wonder if we can offload hook generation to the codegen by providing our own  class (`apolloReactHooksImportFrom `)

https://www.graphql-code-generator.com/docs/plugins/typescript-react-apollo#with-react-hooks